src/Microsoft.TemplateEngine.Edge/DefaultEnvironment.cs

```diff
@@ -18,7 +19,10 @@ public sealed class DefaultEnvironment : IEnvironment
         private const int DefaultBufferWidth = 160;
         private readonly IReadOnlyDictionary<string, string> _environmentVariables;
 
-        public DefaultEnvironment()
+        public DefaultEnvironment() : this(null)
```

Comment:
just a thought - would it be better to do own implementation? Maybe something as `ExtendedEnvironment`?

another thought: https://github.com/dotnet/sdk/blob/main/src/Cli/Microsoft.TemplateEngine.Cli/CliEnvironment.cs we can get rid of this custom implementation with this class now.

Reply:
There were few options:
* Unseal the `DefaultEnvironment` and extend.
* Copy paste `DefaultEnvironment` logic and add custom logic
* Add the logic into `DefaultEnvironment`

The last one felt as the one with least dept incured. Though I might miss some obvious reason why it might not be an optional choice - so I'm open to alternative thoughts :-)